node/src/database/db_initializer.rs

```diff
@@ -496,11 +518,111 @@ pub fn connection_or_panic(
         })
 }
 
+#[derive(Clone)]
+pub struct DbInitializationConfig {
+    pub special_conn_setup: Vec<fn(&Connection) -> rusqlite::Result<()>>,
+    //only things related to db migrations are underneath
+    pub should_be_suppressed: Suppression,
+    pub external_dataset: Option<ExternalData>,
+}
+
+#[derive(PartialEq, Eq, Debug, Clone, Copy)]
+pub enum Suppression {
+    No,
+    Yes,
+    WithErr,
+}
+
+impl DbInitializationConfig {
+    pub fn add_special_conn_setup(
+        mut self,
+        setter: fn(&Connection) -> rusqlite::Result<()>,
+    ) -> Self {
+        self.special_conn_setup.push(setter);
+        self
+    }
+
+    pub fn panic_on_migration() -> Self {
+        Self {
+            special_conn_setup: vec![],
+            should_be_suppressed: Suppression::No,
+            external_dataset: None,
+        }
+    }
+
+    pub fn create_or_migrate(external_params: ExternalData) -> Self {
+        //is used also if a fresh db is being created
+        Self {
+            special_conn_setup: vec![],
+            should_be_suppressed: Suppression::No,
+            external_dataset: Some(external_params),
+        }
+    }
+
+    pub fn migration_suppressed() -> Self {
```

Comment:
Make clear here the difference (if any) between `panic_on_migration()` and `migration_suppressed()`.